container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java

```diff
@@ -275,14 +288,43 @@ private HttpResponse internalServerErrorResponse(HttpRequest request, RuntimeExc
         return errorResponse(request, ErrorMessage.createInternalServerError(Exceptions.toMessageString(e)));
     }
 
-    private HttpSearchResponse handleBody(HttpRequest request) {
+    private HttpResponse invalidJsonResponse(HttpRequest request, JSONException e) {
+        return errorResponse(request, ErrorMessage.createBadRequest(Exceptions.toMessageString(e)));
+    }
+
+    private HttpSearchResponse handleBody(HttpRequest request) throws JSONException {
         // Find query profile
         String queryProfileName = request.getProperty("queryProfile");
         CompiledQueryProfile queryProfile = queryProfileRegistry.findQueryProfile(queryProfileName);
         boolean benchmarkOutput = VespaHeaders.benchmarkOutput(request);
 
         // Create query
-        Query query = new Query(request, queryProfile);
+        Query query;
+        if (request.getMethod() == com.yahoo.jdisc.http.HttpRequest.Method.POST && request.getHeader(com.yahoo.jdisc.http.HttpHeaders.Names.CONTENT_TYPE).equals(JSON_CONTENT_TYPE)) {
+            Inspector inspector = null;
+            try {
+                InputStream inputStream = request.getData();
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                byte[] buffer = new byte[0xFFFF];
+                for (int len = inputStream.read(buffer); len != -1; len = inputStream.read(buffer)) {
+                    outputStream.write(buffer, 0, len);
```

Comment:
`(com.yahoo.io.)IOUtils.readBytes(request.getData(), 1 << 20)` to fill that array ;)